src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -29,7 +29,9 @@ public void Main(string[] args)
             var tempConfig = tempBuilder.Build();
             var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
 
-            var appBasePath = _serviceProvider.GetRequiredService<IApplicationEnvironment>().ApplicationBasePath;
+            var appEnv = _serviceProvider.GetRequiredService<IApplicationEnvironment>();
+            appEnv.SetData("CommandLineArguments",args);
```

Comment:
If the SetData/GetData is a hack and not to be used; it should be marked obsolete; in my opinion otherwise developers will use it.